From 23a9b2c61e592498c9a90ddf5601fd77b25ba3c6 Mon Sep 17 00:00:00 2001
From: Grzegorz Bajorski <grzegorz.bajorski@tieto.com>
Date: Mon, 5 Mar 2012 16:26:19 +0200
Subject: [PATCH 55/58] P2P: Filter input parameters in p2p_serv_disc_external

p2p_ctrl_serv_disc_external() can take only one parameters "0" or "1".

Signed-hostap: Grzegorz Bajorski <grzegorz.bajorski@tieto.com>
(cherry picked from commit 28ef705d2df8002ef1c69a0089cd493f41c2157f)
---
 wpa_supplicant/ctrl_iface.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -2633,6 +2633,8 @@ static int p2p_ctrl_serv_disc_resp(struc
 static int p2p_ctrl_serv_disc_external(struct wpa_supplicant *wpa_s,
 				       char *cmd)
 {
+	if (os_strcmp(cmd, "0") && os_strcmp(cmd, "1"))
+		return -1;
 	wpa_s->p2p_sd_over_ctrl_iface = atoi(cmd);
 	return 0;
 }
